### PR TITLE
Group tags by era (Shelley vs Byron) in swagger documentation

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#774 #775 #776 #777 #778 

# Overview

- [x] I have renamed `/external-transactions` to `/proxy/transactions` to better convey its meaning
- [x] I have renamed all `/byron/wallets` to `/byron-wallets` as per the discussion we had on slack
- [x] I have renamed `/.../migrate` to `/.../migrations`, REST is about resources, not actions. 
- [x] I have grouped and organize API tags by eras such that it's more readable:

Screenshot from 2019-10-16 13-38-08

# Comments

~~:warning: base branch is `jonathanknowles/migrate-byron-wallet-return-type`~~

### CODE REVIEW

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -1196,22 +1199,13 @@ paths:
...
-
-  /byron/wallets:
+  /byron-wallets:
     get:
       operationId: listByronWallets
       tags: ["Byron Wallets"]
       summary: List
       description: |
-        <p align="right">status: <strong>not implemented</strong></p>
+        <p align="right">status: <strong>provisional</strong></p>
```
Comment (piotr-iohk) [inline specifications/api/swagger.yaml]: as discussed... _under testing_ sounds better...

Comment (KtorZ) [reply specifications/api/swagger.yaml]: @piotr-iohk done

**Comment by KtorZ** [comment]
bors try

**Review by piotr-iohk — APPROVED** [review]
:+1:

**Review by jonathanknowles — COMMENTED** [review]
@KtorZ 

I have a some questions about this PR:

1. If `/proxy` refers to a _resource_, then to what resource does it refer?
2. If `/proxy` _doesn't_ refer to a resource, then why can we have `/proxy/transactions` but not `/byron/wallets`?

**Comment by KtorZ** [comment]
@jonathanknowles 

The `proxy` is so-to-speak, the node's client itself. So, we might (conceptually) imagine other endpoint on that resource, like `/proxy/tip` or `/proxy/health`. Doing something like `GET /proxy` probably doesn't make sense as we would have to come to a representation for that `proxy`, but it could be possible. 

`byron` does not refer to anything tangible, it's a type of wallet. `byron` is nothing that can be represented. You could argue that we could have these "era"  resources, and have somethings like:

```
/eras/byron/wallets/...
/eras/shelley/wallets/...
```

but it's a bit far-fetched (and would cause a pretty big changes for all other endpoints to keep the consistency).
**edit:** So to be clear, `byron` or `shelley` acts more as ids here, or discriminant for resources. 


Going further with RESTful practices, I recommend reading: https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#restful

**Comment by KtorZ** [comment]
bors try

**Comment by KtorZ** [comment]
Fixed tests and re-run integration tests locally :+1:, they pass.

**Pull request merged**